lib/rules/no-dupe-disjunctions.ts

```diff
@@ -121,6 +125,27 @@ function containsAssertionsOrUnknowns(
     }
 }
 
+/**
+ * Returns whether the given nodes contains any features that cannot be
+ * expressed by pure regular expression. This mainly includes assertions and
+ * backreferences.
+ */
+function isNonRegular(node: Node): boolean {
+    try {
+        visitRegExpAST(node, {
+            onAssertionEnter() {
+                throw new Error()
+            },
+            onBackreferenceEnter() {
+                throw new Error()
+            },
+        })
+        return false
+    } catch (error) {
+        return true
+    }
```

Comment:
Is it possible to use hasSomeDescendant?

```suggestion
    return hasSomeDescendant(
        node,
        (d) => d.type === "Assertion" || d.type === "Backreference",
    )
```

Reply:
Yes, it is! I should have though of that. Thank you.